include/amici/splinefunctions.h

```diff
@@ -0,0 +1,407 @@
+
+    /**
+     * @brief Compute the coefficients for all polynomial segments of this spline
+     */
+    virtual void compute_coefficients() = 0;
+
+    /**
+     * @brief Compute the coefficients for all polynomial segments of
+     * the derivatives of this spline with respect to the parameters
+     * @param nplist number of parameters
+     * @param spline_offset offset of this spline inside `dvaluesdp`
+     * and `dslopesdp`
+     * @param dvaluesdp derivatives of the spline values with respect to the
+     * parameters (for all splines in the model, not just this one)
+     * @param dslopesdp derivatives of the spline derivatives with respect
+     * to the parameters (for all splines in the model, not just this one)
+     * @remark The contents of `dvaluesdp` and `dslopesdp` may be modified
+     * by this function.
+     */
+    virtual void compute_coefficients_sensi(int nplist,
+                                            int spline_offset,
+                                            gsl::span<realtype> dvaluesdp,
+                                            gsl::span<realtype> dslopesdp) = 0;
+
+    /**
+     * @brief Get the value of this spline at a given point
+     * @param t point at which the spline is to be evaluated
+     * @return value of the spline at `t`
+     */
+    realtype get_value(const realtype t) const;
+
+    /**
+     * @brief Get the value of this spline at a given point
+     * in the scale in which interpolation is carried out (e.g., log-scale)
+     * @param t point at which the spline is to be evaluated
+     * @return scaled value of the spline at `t`
+     */
+    virtual realtype get_value_scaled(const realtype t) const = 0;
+
+    /**
+     * @brief Get the value of this spline at a given node
+     * @param i index of the node at which the spline is to be evaluated
+     * @return value of the spline at the `i`-th node
+     */
+    realtype get_node_value(const int i) const;
+
+    /**
+     * @brief Get the value of this spline at a given node
+     * in the scale in which interpolation is carried out (e.g., log-scale)
+     * @param i index of the node at which the spline is to be evaluated
+     * @return scaled value of the spline at the `i`-th node
+     */
+    realtype get_node_value_scaled(const int i) const;
+
+    /**
+     * @brief Get the derivative of this spline with respect to a given
+     * parameter at a given point
+     * @param t point at which the sensitivity is to be evaluated
+     * @param ip index of the parameter
+     * @return sensitivity of the spline with respect to the `ip`th parameter
+     * at `t`
+     */
+    realtype get_sensitivity(const realtype t, const int ip) const;
+
+    /**
+     * @brief Get the derivative of this spline with respect to a given
+     * parameter at a given point
+     * @param t point at which the sensitivity is to be evaluated
+     * @param ip index of the parameter
+     * @param value value of the spline at the given time point.
```

Comment:
You mean the three argument `get_sensitivity`? If the spline interpolation is computed in log-space, then the final function is actually `f(x; p) = exp(spline(x; p))`. Thus the derivative of `f` with respect to the parameter `p` is `exp(spline(x; p)) * dspline/dp(x; p)`. If we already have the value of `spline(x; p)` (which we have in the AMICI simulation loop) we can pass it to the 3-argument version so that it does not get recomputed.